brave/brave-core
# Move Confirmations into ads process

Resolves https://github.com/brave/brave-browser/issues/9870

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

- Confirm visited http/https pages are classified
- Confirm users are rewarded for viewing ads
- Confirm confirmation is sent to the server when dismissing an ad
- Confirm confirmation is sent to the server when an ad has landed (sustained)
- Confirm confirmation is sent to the server when flagging an ad
- Confirm confirmation is sent to the server when upvoting an ad
- Confirm confirmation is sent to the server when downvoting an ad
- Confirm brave://rewards UI is updated when rewarded for viewing an ad (also confirm same value after restarting the browser)
- Confirm brave://rewards UI is updated when viewing an ad (also confirm same value after restarting the browser)
- Confirm brave://rewards UI is updated when claiming an ads grant (also confirm same value after restarting the browser)
- Confirm "Ad notifications received this month" reset to 0 on the 1st of the month (also confirm same value after restarting the browser)
- Confirm "Next payment date" is working as expected (also confirm same value after restarting the browser)
- Confirm tokens are cashed-out periodically
- Confirm tokens are refilled when running low
- Confirm catalog is downloaded periodically
- Confirm ad conversions are working as expected
- Confirm state level targeting is working as expected
- Confirm purchase intent is working as expected

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
